ember-file-upload/addon/components/file-upload.ts

```diff
@@ -88,11 +91,26 @@ interface FileUploadArgs {
  * }
  * ```
  *
- * @deprecated use `{{file-queue}}` helper with `{{queue.selectFile}}` modifier
+ * @deprecated Use `{{file-queue}}` helper with `{{queue.selectFile}}` modifier.
  */
 export default class FileUploadComponent extends Component<FileUploadArgs> {
   @service declare fileQueue: FileQueueService;
 
+  constructor(owner: unknown, args: FileUploadArgs) {
+    super(owner, args);
+
+    deprecate(
+      `\`<FileUpload>\` is deprecated. Use \`{{file-queue}}\` helper with \`{{queue.selectFile}}\` modifier.`,
+      false,
+      {
+        for: 'ember-file-upload',
+        id: 'file-upload',
+        since: { enabled: 'v5.0.0' },
+        until: 'v6.0.0',
+      }
+    );
```

Comment:
There's a lot of deprecated arguments also – but I figured a single deprecation warning with a (future) link to good upgrade docs is the best way forward here.